internal/gatewayapi/translator.go

```diff
@@ -234,13 +235,21 @@ func (t *Translator) InitIRs(gateways []*GatewayContext, resources *Resources) (
 	for _, gateway := range gateways {
 		gwXdsIR := &ir.Xds{}
 		gwInfraIR := ir.NewInfra()
+		labels := infrastructureLabels(gateway.Gateway)
```

Comment:
how does this interact with pod labels and annotations defined in https://gateway.envoyproxy.io/v0.6.0/api/extension_types/#kubernetespodspec as well as service annotations 
https://gateway.envoyproxy.io/v0.6.0/api/extension_types/#kubernetesservicespec ?

Reply:
same key in envoyproxy take first

Reply:
is it appended ? can we had a test case for it ?

Reply:
it's a map, so.

yes, I will add a test for it.